phpBB/phpbb/composer/installer.php

```diff
@@ -0,0 +1,695 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\composer;
+
+use Composer\Composer;
+use Composer\Factory;
+use Composer\IO\IOInterface;
+use Composer\IO\NullIO;
+use Composer\Json\JsonFile;
+use Composer\Package\BasePackage;
+use Composer\Package\CompletePackage;
+use Composer\Package\PackageInterface;
+use Composer\Repository\ComposerRepository;
+use Composer\Repository\RepositoryInterface;
+use Composer\Semver\Constraint\ConstraintInterface;
+use Composer\Util\RemoteFilesystem;
+use phpbb\config\config;
+use phpbb\exception\runtime_exception;
+
+/**
+ * Class to install packages through composer while freezing core dependencies.
+ */
+class installer
+{
+	const PHPBB_TYPES = 'phpbb-extension,phpbb-style,phpbb-language';
+
+	/**
+	 * @var array Repositories to look packages from
+	 */
+	protected $repositories = [];
+
+	/**
+	 * @var bool Indicates whether packagist usage is allowed or not
+	 */
+	protected $packagist = false;
+
+	/**
+	 * @var string Composer filename used to manage the packages
+	 */
+	protected $composer_filename = 'composer-ext.json';
+
+	/**
+	 * @var string Directory where to install packages vendors
+	 */
+	protected $packages_vendor_dir = 'vendor-ext/';
+
+	/**
+	 * @var string Minimum stability
+	 */
+	protected $minimum_stability = 'stable';
+
+	/**
+	 * @var string phpBB root path
+	 */
+	protected $root_path;
+
+	/**
+	 * @var string Stores the original working directory in case it has been changed through move_to_root()
+	 */
+	private $original_cwd;
+
+	/**
+	 * @var array Stores the content of the ext json file before generate_ext_json_file() overrides it
+	 */
+	private $ext_json_file_backup;
+
+	/**
+	 * @param \phpbb\config\config	$config		Config object
+	 * @param string				$root_path	phpBB root path
+	 */
+	public function __construct($root_path, config $config = null)
+	{
+		if ($config)
+		{
+			$repositories        = json_decode($config['exts_composer_repositories'], true);
```

Comment:
Another thing I think we should get rid of in coding guidelines, inline indentation really only causes overly complicated diffs later on.